spring-kafka/src/main/java/org/springframework/kafka/support/serializer/DelegatingByTopicSerializer.java

```diff
@@ -0,0 +1,245 @@
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * A {@link Serializer} that delegates to other serializers based on a topic pattern.
+ *
+ * @author Gary Russell
+ * @since 2.8
+ *
+ */
+public class DelegatingByTopicSerializer implements Serializer<Object> {
+
+	private static final LogAccessor LOGGER = new LogAccessor(DelegatingDeserializer.class);
+
+	/**
+	 * Name of the configuration property containing the serialization selector map for
+	 * values with format {@code selector:class,...}.
+	 */
+	public static final String VALUE_SERIALIZATION_TOPIC_CONFIG = "spring.kafka.value.serialization.bytopic.config";
+
+	/**
+	 * Name of the configuration property containing the serialization topic pattern map for
+	 * keys with format {@code pattern:class,...}.
+	 */
+	public static final String KEY_SERIALIZATION_TOPIC_CONFIG = "spring.kafka.key.serialization.bytopic.config";
+
+	/**
+	 * Set to false to make topic pattern matching case-insensitive.
+	 */
+	public static final String CASE_SENSITIVE = "spring.kafka.value.serialization.bytopic.case.insensitive";
+
+	private final Map<Pattern, Serializer<?>> delegates = new ConcurrentHashMap<>();
+
+	private boolean forKeys;
+
+	private boolean cased = true;
+
+	/**
+	 * Construct an instance that will be configured in {@link #configure(Map, boolean)}
+	 * with producer properties {@link #VALUE_SERIALIZATION_TOPIC_CONFIG} and
+	 * {@link #KEY_SERIALIZATION_TOPIC_CONFIG}.
+	 */
+	public DelegatingByTopicSerializer() {
+	}
+
+	/**
+	 * Construct an instance with the supplied mapping of topic patterns to delegate
+	 * serializers.
+	 * @param delegates the map of delegates.
+	 */
+	public DelegatingByTopicSerializer(Map<Pattern, Serializer<?>> delegates) {
+		this.delegates.putAll(delegates);
+	}
+
+	/**
+	 * Set to false to make topic name matching case insensitive.
+	 * @param caseSensitive false for case insensitive.
+	 */
+	public void setCaseSensitive(boolean caseSensitive) {
+		this.cased = caseSensitive;
+	}
+
+	@Override
+	public void configure(Map<String, ?> configs, boolean isKey) {
+		this.forKeys = isKey;
+		Object insensitive = configs.get(CASE_SENSITIVE);
+		if (insensitive instanceof String) {
+			this.cased = Boolean.parseBoolean((String) insensitive);
+		}
+		String configKey = configKey();
+		Object value = configs.get(configKey);
+		if (value == null) {
+			return;
+		}
+		else if (value instanceof Map) {
+			processMap(configs, isKey, configKey, value);
+		}
+		else if (value instanceof String) {
+			this.delegates.putAll(createDelegates((String) value, configs, isKey));
+		}
+		else {
+			throw new IllegalStateException(
+					configKey + " must be a map or String, not " + value.getClass());
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private void processMap(Map<String, ?> configs, boolean isKey, String configKey, Object value) {
```

Comment:
I think the `value` can directly be a `Map<Object, Object>`. You don't call this method unless `value instanceof Map`.
Am I missing anything specific?